src/tools/common/sss_tools.c

```diff
@@ -495,6 +507,11 @@ errno_t sss_tool_popt_ex(struct sss_cmdline *cmdline,
 done:
     poptFreeContext(pc);
     talloc_free(help);
+    if (ret != EOK && _fopt != NULL) {
+        free(discard_const(*_fopt));
+        *_fopt = NULL;
```

Comment:
`talloc_zfree()`?

Reply:
Do you mean for `*_fopt`? Its memory is currently allocated with `strdup`, I could change that but I don't see any benefit taking into account that we aren't building any hierarchical memory tree.

Reply:
Ah, my mistake, sorry.